src/bin/pg_rewind/parsexlog.c

```diff
@@ -356,6 +360,11 @@ SimpleXLogPageRead(XLogReaderState *xlogreader, XLogRecPtr targetPagePtr,
 	 */
 	Assert(xlogreadfd != -1);
 
+#ifdef PERCONA_EXT
+	r = xlog_smgr->seg_read(xlogreadfd, readBuf, XLOG_BLCKSZ, (off_t) targetPageOff,
+							targetHistory[private->tliIndex].tli,
+							xlogreadsegno, WalSegSz);
+#else
 	/* Read the requested page */
```

Comment:
Maybe this comment could go before the ifdef?

Reply:
done